packages/smart/funcs.go

```diff
@@ -43,6 +43,7 @@ import (
 // SmartContract is storing smart contract data
 type SmartContract struct {
 	VDE           bool
+	NotRollback   bool
```

Comment:
Use of inverted logic, maybe it would be better to rename field to Rollback, and do && sc.Rollback, instead of && !sc.NotRollback